src/components/Combobox/ComboBoxElements.tsx

```diff
@@ -0,0 +1,472 @@
+import { mergeRefs } from "@/utils/mergeRefs";
+import {
+  FocusEvent,
+  HTMLAttributes,
+  MouseEvent,
+  ReactNode,
+  forwardRef,
+  useEffect,
+  useRef,
+} from "react";
+import styled from "styled-components";
+import { GenericMenuItem } from "../GenericMenu";
+import { Icon, HorizontalDirection, IconName, Separator, IconButton } from "@/components";
+import { useCombobox } from "./useCombobox";
+import { ComboboxItemProps } from "./types";
+
+import * as RadixPopover from "@radix-ui/react-popover";
+import { IconSize } from "../Icon/types";
+import { EllipsisContainer } from "../commonElement";
+
+declare type DivProps = HTMLAttributes<HTMLDivElement>;
+interface GroupProps extends Omit<DivProps, "value" | "heading"> {
+  heading?: ReactNode;
+  value?: string;
+}
+
+const ComboboxGroupContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: center;
+  width: -webkit-fill-available;
+  width: fill-available;
+  width: stretch;
+  overflow: hidden;
+  background: transparent;
+  &[aria-selected] {
+    outline: none;
+  }
+
+  ${({ theme }) => `
+    font: ${theme.click.genericMenu.item.typography.sectionHeader.default};
+    color: ${theme.click.genericMenu.item.color.text.muted};
+  `};
+  [cui-combobox-group-heading] {
+    display: flex;
+    width: 100%;
+    flex-direction: column;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    ${({ theme }) => `
+     font: ${theme.click.genericMenu.item.typography.sectionHeader.default};
+     color: ${theme.click.genericMenu.item.color.text.muted};
+     padding: ${theme.click.genericMenu.sectionHeader.space.top} ${theme.click.genericMenu.item.space.x} ${theme.click.genericMenu.sectionHeader.space.bottom};
+     gap: ${theme.click.genericMenu.item.space.gap};
+     border-bottom: 1px solid ${theme.click.genericMenu.item.color.stroke.default};
+   `}
+  }
+  &[hidden] {
+    display: none;
+  }
+`;
+
+const ComboboxGroupContent = styled.div`
+  width: inherit;
+`;
+
+export const ComboboxGroup = forwardRef<HTMLDivElement, GroupProps>(
+  ({ children, heading, ...props }, forwardedRef) => {
+    useCombobox(); // inorder to refresh the content on search
```

Comment:
@serdec @fneves @gjones Do you think I should create a hook as below inorder for this component to be rerendered only on search?

import { useMemo } from "react";
import { useCombobox } from "./useCombobox";

```
const useSearch = () => {
  const { search } = useCombobox();
  const searchValue = useMemo(() => search, [search]);
  return searchValue;
};
export default useSearch;

```